emote/memory/storage.py

```diff
@@ -202,6 +202,48 @@ def with_only_last(storage, shape, dtype):
         return NextElementMapper(storage, shape, dtype, only_last=True)
 
 
+class NextNElementWrapper(VirtualStorage):
+    """Simple mapper that can be used to sample a specified N steps over"""
+
+    class Wrapper:
+        def __init__(self, item, n: int):
+            self._item = item
+            self._n = n
+
+        def __getitem__(self, key):
+            if isinstance(key, int):
+                key += self._n
+            elif isinstance(key, tuple):
+                key = tuple(k + self._n for k in key)
+            elif isinstance(key, slice):
+                key = slice(key.start + self._n, key.stop + self._n, key.step)
```

Comment:
Maybe raise here if it's not any of these?

Reply:
Sure!